panels/user-accounts/um-app-permissions.c

```diff
@@ -134,19 +134,52 @@ static const gchar * const oars_categories[] =
 
 /* Auxiliary methods */
 
+static gint
+app_compare_id_length_cb (gconstpointer a,
+                          gconstpointer b)
+{
+  GAppInfo *info_a = (GAppInfo *) a, *info_b = (GAppInfo *) b;
+  const gchar *id_a, *id_b;
+
+  id_a = g_app_info_get_id (info_a);
+  id_b = g_app_info_get_id (info_b);
+
+  if (id_a == NULL && id_b == NULL)
+    return 0;
+  else if (id_a == NULL)
+    return -1;
+  else if (id_b == NULL)
+    return 1;
```

Comment:
It's a shame one can't use:

```c
if (id_a == NULL || id_b == NULL)
  return id_a - id_b;
```

because pointer arithmetic returns a type wider than `int`... :-)

Reply:
I think that’s not particularly clear/obvious to read.

Reply:
Not a serious suggestion!